### PR TITLE
Set `logs_config.cca_in_ad: true` by default

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Enables a feature flag by default.

### Motivation

This feature has not changed since 7.39, where we have been incrementally enabling it and will continue to do so.   Assuming no further issues are found during 7.39 testing, nor during 7.40 QA, this will enable the flag by default for all customers in the 7.40 release.

### Describe how to test/QA your changes

7.39 QA of the relevant functionality set this flag to true, so no additional functionality-specific QA is required.  Other agent QA may uncover unanticipated consequences of this change.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+other:
+  - |
+    The ``logs_config.cca_in_ad`` feature flag now defaults to true.  This
```
Comment (cswatt) [inline releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml:11]: is this supposed to have double backticks?

File: releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    selects updated codepaths in autodiscovery and the logs-agent.  No behavior
```
Comment (cswatt) [inline releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml:12]:
```suggestion
    selects updated codepaths in Autodiscovery and the Logs Agent.  No behavior
```

File: releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml (hunk 1 above)
Comment (djmitche) [reply releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml:11]: Yes, it's reStructuredText.

**Review by hush-hush — APPROVED** [review]

**Comment by djmitche** [comment]
Oh, I bet those test failures _are_ related, as the defaults cause random fluctuations in expected values :(

**Review by gh123man — APPROVED** [review]

**Review by juliogreff — APPROVED** [review]

**Pull request merged**